src/python/pants/backend/plugin_development/pants_requirements.py

```diff
@@ -61,40 +73,20 @@ class GenerateFromPantsRequirementsRequest(GenerateTargetsRequest):
     generate_from = PantsRequirementsTargetGenerator
 
 
-def determine_version() -> str:
-    # Because the Plugin API is not stable, it can have breaking changes in-between dev releases.
-    # Technically, it can also have breaking changes between rcs in the same release series, but
-    # this is much less likely.
-    #
-    # So, we require exact matches when developing against a dev release, but only require
-    # matching the release series if on an alpha release, rc, or stable release.
-    #
-    # If this scheme does not work for users, they can:
-    #
-    #    1. Use a `python_requirement` directly
-    #    2. Add a new `version` field to this target generator.
-    #    3. Fork this target generator.
-    return (
-        f"=={PANTS_SEMVER}"
-        if PANTS_SEMVER.is_devrelease
-        else (
-            f">={PANTS_SEMVER.major}.{PANTS_SEMVER.minor}.0a0,"
-            f"<{PANTS_SEMVER.major}.{PANTS_SEMVER.minor + 1}"
-        )
-    )
-
-
 @rule
 def generate_from_pants_requirements(
     request: GenerateFromPantsRequirementsRequest, union_membership: UnionMembership
 ) -> GeneratedTargets:
     generator = request.generator
-    version = determine_version()
+    version = generator[PantsRequirementsVersionField].value
 
     def create_tgt(dist: str, module: str) -> PythonRequirementTarget:
         return PythonRequirementTarget(
             {
-                PythonRequirementsField.alias: (f"{dist}{version}",),
+                PythonRequirementsField.alias: (f"{dist} == {version}",),
+                PythonRequirementFindLinksField.alias: (
+                    f"https://github.com/pantsbuild/pants/releases/expanded_assets/release_{version}",
```

Comment:
Awesome find! Two questions:

1. I assume the host-less URLs (e.g. `<a href="/pantsbuild/pants/releases/download/release_2.16.0/pantsbuild.pants-2.16.0-cp39-cp39-manylinux2014_x86_64.whl" ...>`) work?
2. How stable is it? I could only find references like https://www.reddit.com/r/github/comments/y5qfaz/comment/j6uzlmc and no explicit docs from Github

Reply:
For 1. Yes. Or at least, my local testing says yes :sweat_smile: 

But Let's go down the `pip` rabbit hole: https://github.com/pypa/pip/blob/593b85f4abd30688648436bb9baca3b8f7b32b51/src/pip/_internal/models/link.py#L327C47-L327C54

```python
$ python
Python 3.8.10 (default, May 26 2023, 14:05:08) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib.parse
>>> urllib.parse.urljoin("https://foo.bar", "https://foo.bar/baz.whl")
'https://foo.bar/baz.whl'
>>> urllib.parse.urljoin("https://foo.bar", "/baz.whl")
'https://foo.bar/baz.whl'
```

For 2. I don't think I have a concrete answer :worried: We can't point to the release page because the asset list is populated using JS with a query to this URL. So just `--find-links` to the release page yields nada. It certainly doesn't scream private, but that doesn't say much because there isn't a real standard or convention for URLs.

I opened a [public discussion](https://github.com/orgs/community/discussions/61347), and we have some work connections with GitHub so I'll flex there and see if there's an answer.

Reply:
OK, so from my chatting with GitHub, since it isn't publicly documented we shouldn't rely on it.

I'll have to whip up a new PR that manages an index on S3 with ALL our wheels on it, and we can point to that,